src/NodeApi/JSValue.cs

```diff
@@ -391,29 +391,58 @@ public unsafe BigInteger ToBigInteger()
 
     public bool IsBigInt() => TypeOf() == JSValueType.BigInt;
 
-    public bool Is<TValue>() where TValue : struct
+    /// <summary>
+    /// Checks if the T struct can be created from this `JSValue`.
+    /// </summary>
+    /// <typeparam name="T">A struct that implements IJSValue interface.</typeparam>
+    /// <returns>
+    /// `true` if the T struct can be created from this `JSValue`. Otherwise it returns `false`.
+    /// </returns>
+    public bool Is<T>() where T : struct, IJSValue<T>
```

Comment:
Should these methods (`Is<>`, `As<>`, `AsUnchecked<>`, `CastTo<>`) be default interface implementations of `IJSValue<>` instead of methods of `JSValue`? Then they could convert _between_ specific types, not only from `JSValue` to a specific type. For example it is valid to convert a `JSArray` or `JSPromise` to `JSObject`.

Reply:
This assumes `JSValue` also implements `IJSValue<JSValue>` as suggested above.

Reply:
Unfortunately the default interface implementations are not accessible from the struct or classes that implement it. I have added these methods to all structs and now we can convert between all types that implement `IJSValue`.
As a result, the `AsJSValue()` method became redundant and I had removed it.